### PR TITLE
Logging of generated macros code 

### PR DESCRIPTION
**Pull request by jasiak**
Added logging of generated macros code when env var TAPIR_LOG_GENERATED_CODE = true

### CODE REVIEW

File: core/src/main/scala/tapir/generic/Debug.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  import scala.reflect.macros.blackbox
+  import scala.reflect.runtime.universe.{TypeTag, typeOf}
+
+  private val macroDebugEnabled = System.getenv("TAPIR_LOG_GENERATED_CODE") == "true"
+
+  def logGeneratedCode[E: c.WeakTypeTag](c: blackbox.Context)(tree: c.universe.Tree): Unit = {
+    if (macroDebugEnabled) {
+      println(s"""Generated schema code for ${name[E]}:""")
+      println(tree)
```
Comment (ghostbuster91) [inline core/src/main/scala/tapir/generic/Debug.scala]:
```suggestion
      println(showCode(tree))
```

File: core/src/main/scala/tapir/generic/Debug.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+object Debug {
+
+  import scala.reflect.macros.blackbox
+  import scala.reflect.runtime.universe.{TypeTag, typeOf}
+
+  private val macroDebugEnabled = System.getenv("TAPIR_LOG_GENERATED_CODE") == "true"
+
+  def logGeneratedCode[E: c.WeakTypeTag](c: blackbox.Context)(tree: c.universe.Tree): Unit = {
+    if (macroDebugEnabled) {
+      println(s"""Generated schema code for ${name[E]}:""")
```
[the remark this answers is not in the document]
Comment (ghostbuster91) [reply core/src/main/scala/tapir/generic/Debug.scala]: How about `${name[R]} macro output` and closing it with something similar?

File: core/src/main/scala/tapir/generic/Debug.scala (hunk 1 above)
Comment (jasiak) [reply core/src/main/scala/tapir/generic/Debug.scala]: Fixed

**Comment by adamw** [comment]
Thanks! :)

**Pull request merged**